src/main/java/com/anaplan/client/AnaplanService.java

```diff
@@ -0,0 +1,228 @@
+package com.anaplan.client;
+
+import com.anaplan.client.ex.ActionsNotFoundException;
+import com.anaplan.client.ex.ProcessesNotFoundException;
+import com.anaplan.client.ex.ServerFilesNotFoundException;
+import com.google.common.base.Strings;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wrapper class of the Anaplan service providing the interfaces for a client to connect with
+ * Anaplan server.
+ */
+public class AnaplanService implements Serializable {
+
+  /** Enum class of the Anaplan internal function types. */
+  public enum FunctionType {
+    PROCESS,
+    ACTION
+  }
+
+  /** Interface for the CDF plugin that use Anaplan service. */
+  public interface AnaplanConfig {
+    String getServiceLocation();
+
+    String getAuthServiceLocation();
+
+    String getUsername();
+
+    String getPassword();
+  }
+
+  public static final String NAME_SERVICE_LOCATION = "serviceLocation";
+  public static final String NAME_AUTH_SERVICE_LOCATION = "authServiceLocation";
+  public static final String NAME_USERNAME = "username";
+  public static final String NAME_PASSWORD = "password";
+  public static final String NAME_WORKSPACE_ID = "workspaceId";
+  public static final String NAME_MODEL_ID = "modelId";
+  public static final String NAME_SERVER_FILE_NAME = "serverFileName";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnaplanService.class);
+
+  /**
+   * Sets metadata details for the connection with Anaplan server
+   *
+   * @param config the config instance carries metadata for the connection with Anaplan server
+   */
+  public static void setTaskMetadata(AnaplanConfig config) {
+    setAPIRoot(config.getServiceLocation(), config.getAuthServiceLocation());
+    setCredential(config.getUsername(), config.getPassword());
+  }
+
+  private static void setAPIRoot(String serviceLocation, String authServiceLocation) {
+    try {
+      Program.setServiceLocation(new URI(serviceLocation));
+      Program.setAuthServiceLocation(new URI(authServiceLocation));
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e.getMessage());
```

Comment:
catching a checked exception and rethrowing as a RuntimeException is usually a sign that errors are not being handled in the right place. 

In this case, it's better if the AnaplanConfig's getter methods return URI instead of String, so that anything using AnaplanConfig doesn't have to handle invalid values. Parsing strings into URI should happen as early as possible, where a better error message can be thrown. For example, if these are user provided values, they should be validated at a higher level, before it reaches this level.

Reply:
Thanks for the instruction! Updated!